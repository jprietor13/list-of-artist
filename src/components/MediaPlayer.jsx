import React from 'react'
import { convertMiliSeconds } from '../utils/utils'

export const MediaPlayer = ({ duration_ms, name, preview_url }) => {
 
  return (
    <>
      <div>
        <audio src={preview_url} controls autoPlay/>
      </div>
      <div>
        <p>{duration_ms && convertMiliSeconds(duration_ms)}</p>
        <p>{name}</p>   
      </div>
    </>
  )
}
