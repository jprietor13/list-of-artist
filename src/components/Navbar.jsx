import React from 'react'

export const Navbar = () => {
  return (
    <div>
      <h1>Prueba</h1>
    </div>
  )
}
