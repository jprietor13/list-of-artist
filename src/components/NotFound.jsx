import React from 'react'
import { NavLink } from 'react-router-dom'

const NotFound = () => {
  return (
    <>
      <NavLink to={"/"} className="return return-notFound">
        Regresar
      </NavLink>
      <h3 className="listArtist__title">Lista de Artistas</h3>
      <h1 className='title-notFound'>Page Not Found</h1>
    </>
  )
}

export default NotFound