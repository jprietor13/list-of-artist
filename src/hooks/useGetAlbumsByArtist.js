import { useEffect, useState } from 'react'
import { getAlbumsByArtist } from '../services/getAlbumsByArtist'

export const useGetAlbumsByArtist = (id) => {
  const [albums, setAlbums] = useState([])

  useEffect(() => {
    getAlbumsByArtist(id)
    .then((data) => {
      setAlbums(data)
    })
  }, [id])

  return { albums }
}
