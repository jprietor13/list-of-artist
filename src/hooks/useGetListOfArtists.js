import { useState, useEffect } from 'react';
import { getListOfArtists } from '../services/getListOfArtists';

export const useGetListOfArtists = () => {

  const [artists, setArtist] = useState([]);

  const getData = async () => {
    const data = await getListOfArtists();
    setArtist(data)
  }

  useEffect(() => {
    getData()
  }, [])

  return artists 
}
