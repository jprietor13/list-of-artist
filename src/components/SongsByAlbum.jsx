import React from 'react'
import { useParams } from 'react-router-dom'
import { useGetAlbumsByArtist } from '../hooks/useGetAlbumsByArtist';
import { useGetSongsByAlbums } from '../hooks/useGetSongsByAlbums';
import { convertMiliSeconds, filterByReference } from '../utils/utils';

export const SongsByAlbum = () => {

  const { idArtist, idAlbum } = useParams();
  const { albums } = useGetAlbumsByArtist(idAlbum);
  const { songs } = useGetSongsByAlbums(idAlbum);

  const filterArtist = albums.filter(item => {
    return item.artist == idArtist
  })

  const getAlbum = filterArtist[0]?.albums.find(item => {
    return item.id == idAlbum
  })

  const songsByAlbum = songs.find(item => {
    return item.album == idAlbum
  })

  const albumsOfArtist = filterArtist.map(item => {
    return item.albums
  })

  const suggestions = filterByReference(songs, albumsOfArtist);

  return (
    <>
      <div>
        <img src={getAlbum?.image} alt={getAlbum?.name} />
        <p>{getAlbum?.name}</p>
      </div>
      <div>
        <h1>Canciones</h1>
        <ul>
          {songsByAlbum?.songs?.map(item => 
            <li key={item.id}>
              <p>{item.name}</p>
              <p>{ convertMiliSeconds(item.duration_ms) }</p>
            </li>
          )}
        </ul>
        <h1>Sugerencias</h1>
        <ul>
          {suggestions.map((item, index) => 
            <li key={index}>
              {item.map(subitem =>
                <div key={subitem.id}>
                  <p>{subitem.name}</p>
                  <p>{convertMiliSeconds(subitem.duration_ms)}</p>
                </div> 
              )}
            </li>  
          )}
        </ul>
      </div>
    </>
  )
}
