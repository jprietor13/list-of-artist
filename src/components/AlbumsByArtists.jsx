import React from "react";
import { useParams, NavLink } from "react-router-dom";
import { useGetAlbumsByArtist } from "../hooks/useGetAlbumsByArtist";
import { useGetListOfArtists } from "../hooks/useGetListOfArtists";

export const AlbumsByArtists = () => {
  const { idArtist } = useParams();
  const { albums } = useGetAlbumsByArtist(idArtist);
  const artists = useGetListOfArtists();

  const artist = artists?.find((item) => {
    return item.id == idArtist;
  });

  const album = albums?.find((item) => {
    return item.artist == idArtist;
  });

  return (
    <>
      <NavLink to={"/"} className="return">
        Regresar
      </NavLink>
      <div>
        <img src={artist?.image} alt={artist?.name} />
        <h2>{artist?.name}</h2>
      </div>
      <div>
        <h2>Albumes</h2>
        {album?.albums.map((item) => (
          <NavLink to={`album/${item.id}`} key={item.id}>
            <div key={item.id}>
              <img src={item.image} alt={item.name} />
              <p>
                {item.name}
              </p>
            </div>
          </NavLink>
        ))}
      </div>
    </>
  );
};
