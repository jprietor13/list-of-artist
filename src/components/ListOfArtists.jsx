import React from "react";
import { NavLink } from "react-router-dom";
import { useGetListOfArtists } from "../hooks/useGetListOfArtists";

export const ListOfArtists = () => {
  const artists = useGetListOfArtists();

  return (
    <div>
      {artists.map((artist, index) => (
        <div key={index}>
          <NavLink to={`/artist/${artist.id}`}>
            <div>
              <img src={artist.image} alt={artist.name} />
              <p>{artist.name}</p>
            </div>
          </NavLink>
        </div>
      ))}
    </div>
  );
};
