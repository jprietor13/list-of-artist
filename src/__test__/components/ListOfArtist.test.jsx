import { React } from "react";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { ListOfArtists } from "../../components/ListOfArtists";
import { useGetListOfArtists } from "../../hooks/useGetListOfArtists";

jest.mock("../../hooks/useGetListOfArtists");

describe("Pruebas en <ListOfArtists />", () => {
  configure({ adapter: new Adapter() });

  test("Debe cargarse correctamente", () => {
    useGetListOfArtists.mockReturnValue([]);
    const wrapper = shallow(<ListOfArtists />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar la informacion", () => {
    useGetListOfArtists.mockReturnValue([
      {
        name: "Artic Monkeys",
      },
    ])
    const wrapper = shallow(<ListOfArtists />);
    
    expect(wrapper.find("p").text().trim()).toBe("Artic Monkeys");
  });
});
