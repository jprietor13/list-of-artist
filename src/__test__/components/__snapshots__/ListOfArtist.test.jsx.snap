// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ListOfArtists /> Debe cargarse correctamente 1`] = `ShallowWrapper {}`;
